concordium-contracts-common/src/schema.rs

```diff
@@ -229,6 +264,32 @@ impl FunctionV2 {
     }
 }
 
+/// Describes the schema of an init or a receive function for V1 contracts with
+/// V3 schemas. Differs from [`FunctionV2`] in that a schema for the events can
+/// be included.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FunctionV3 {
+    pub index:        u8,
```

Comment:
Is this field ever needed?
It seems redundant, since the other fields defines its value